src/main/java/com/microsoft/graph/http/DefaultHttpProvider.java

```diff
@@ -401,9 +407,17 @@ private InputStream handleBinaryStream(final InputStream in) {
      * @param clazz           the type of the response object
      * @return                the JSON object
      */
-    private <Result> Result handleEmptyResponse(Map<String, List<String>> responseHeaders, final Class<Result> clazz) {
+    private <Result> Result handleEmptyResponse(Map<String, List<String>> responseHeaders, final Class<Result> clazz) 
+    		throws UnsupportedEncodingException{
     	//Create an empty object to attach the response headers to
-        InputStream in = new ByteArrayInputStream("{}".getBytes());
+    	InputStream in = null;
+    	try{
+        	in = new ByteArrayInputStream("{}".getBytes(ENCODING_TYPE));
+        }
+        catch(UnsupportedEncodingException ex) {
+        	logger.logError(ex.getMessage(), ex);
```

Comment:
Instead of logging error here, maybe you should create new catch block at the end of sendRequestInternal method. There is already few catch blocks which will log incorrectly in this particular case.